### PR TITLE
User history API

### PR DESCRIPTION
**Pull request by phamann**
Add initial endpoints to query a users tag history from DynamoDB.

- `/__user-prefs/history/:uuid`
- `/__user-prefs/history/:uuid/getSinceDate/:date`
- `/__user-prefs/history/getAllSinceDate/:date`
- Formats the data return from Dynamo:
    - Flatten all sets into single array
    - Counts by occurrence
    - Parses JSON to extract name and id as separate fields
    - Sort by date
- Adds the ability to query using relative day, i.e.: `-4d`

#### Todo
- [ ] Add taxonomy endpoint
- [ ] Tests :smile: 

/cc @adgad @wheresrhys

### CODE REVIEW

File: server/controllers/history/get.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*jshint node:true*/
+
+'use strict';
+
+var History = require('../../models/tables/history');
+var formatHistory = require('../../lib/format-history');
+
+module.exports = function(req, res, next) {
+	var filters = {
+		limit: req.query.limit
```
Comment (adgad) [inline server/controllers/history/get.js]: Shall we add some caching on these routes? Probably doesn't need to be as up to date as some of the others...

**Comment by adgad** [comment]
:+1: nice!

File: server/controllers/history/get.js (hunk 1 above)
Comment (wheresrhys) [inline server/controllers/history/get.js]: :+1:

File: server/lib/format-history.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	.flatten()
+	.countBy()
+	.map(function(value, key){
+		//HACK:
+		//Temporary solution until old data is deleted.
+		try {
+			key = JSON.parse(key);
+			return {
+				id: key.id,
+				name: key.name,
```
Comment (wheresrhys) [inline server/lib/format-history.js]: what was the decision on including taxonomy here?

Comment (phamann) [inline server/lib/format-history.js]: Need to add the the data model in the next-spoor-ingest client first which pushes the data to Dynamo. But yes it should be in here by the end of the day :smile:

File: server/controllers/history/get.js (hunk 1 above)
Comment (phamann) [inline server/controllers/history/get.js]: massive :+1:

**Pull request merged**